File: java/src/main/java/demo/XStreamImport.java
package demo;

import com.thoughtworks.xstream.XStream;
import javax.servlet.http.HttpServletRequest;
import org.springframework.web.bind.annotation.GetMapping;

public final class XStreamImport {
  @GetMapping("/java/xstream/import")
  public Object importXml(HttpServletRequest request) {
    String xml = request.getParameter("xml");
    return parseXml(xml);
  }

  private Object parseXml(String xml) {
    return new XStream().fromXML(xml);
  }
}

File: kotlin/src/main/kotlin/demo/XStreamImport.kt
package demo

import com.thoughtworks.xstream.XStream
import javax.servlet.http.HttpServletRequest
import org.springframework.web.bind.annotation.GetMapping

class XStreamImport {
    @GetMapping("/kotlin/xstream/import")
    fun importXml(request: HttpServletRequest): Any {
        val xml = request.getParameter("xml")
        return parseXml(xml)
    }

    private fun parseXml(xml: String): Any {
        return XStream().fromXML(xml)
    }
}
